### ISSUE TITLE
[1.6.1] eth_getTransactionReceipt 's  transactionIndex != log.transactionIndex

### ISSUE THREAD

**Issue by gpBlockchain** [nervosnetwork/godwoken-web3#495]
## **Version**
- Godwoken v1 
- (Optional) Get the versions from `poly_version` RPC
```
curl --location --request POST 'https://godwoken-alphanet-v1.ckbapp.dev' \
--header 'Content-Type: application/json' \
--data-raw '{
        "jsonrpc":"2.0",
        "method":"poly_version",              
        "params":[
        ],
        "id":1
}'

{"jsonrpc":"2.0","id":1,"result":{"versions":{"web3Version":"1.6.1-rc1","web3IndexerVersion":"1.6.1-rc1","godwokenVersion":"1.5.0-rc2 c36b022"},"nodeInfo":{"nodeMode":"fullnode","rollupCell":{"typeHash":"0x0a6d3bb9392242c0357fcae7ccead8f87b428ed280909bfdfaa549529b85df38","typeScript":{"code_hash":"0x56abab7961e8348aed629a0e59c05d0f6b555314f8f95606eae4bcb2adafdce3","hash_type":"type","args":"0x749f79c58129fb18ce425e030e23f127fe60979ef8f69c28a945f4da19fec591"}},"rollupConfig":{"requiredStakingCapacity":"0x2d79883d2000","challengeMaturityBlocks":"0x1c2","finalityBlocks":"0x64","rewardBurnRate":"0x32","chainId":"0x315de"},"gwScripts":{"deposit":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0xae79482d7b4e639d2ff323465821686344e3e19d0d9223b03297ffb0204fb8a1"},"typeHash":"0x9aa15280cf2123755516ff93180ab14b66a043562ffd70a0947afe7a12d573e5"},"withdraw":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x5d4b16e48cd79b8fcbe5ddcff756e8ebebddd9019d7eb24f5333cc6c26bcaa12"},"typeHash":"0xb580c581b8789dbd3add6c8d2efc63af3590b1a272d950e7dd1c2f64f5b425b5"},"stateValidator":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x514a98ddd783a10ef84025e2888b0d2ce22d040d9d0e0b2fa85cea3436d7dbe6"},"typeHash":"0x56abab7961e8348aed629a0e59c05d0f6b555314f8f95606eae4bcb2adafdce3"},"stakeLock":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0xce25c4130db0c7b3ab2c5c5a5e5c73f1f0e682387d0dddb7509be7dabd168c06"},"typeHash":"0xfeac11dee7c66e54e4864c177215d0cbb6ebcfbd2fe8e0d2bc45b3413542e2da"},"custodianLock":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0xe0a9466ceec538c94e92edc9759a68a2b38f48348a69de920e2841a0239e0497"},"typeHash":"0x38e7fa830764ad269f63fa9e874b39d13d8f4ca42c47cac20e11c847dc5595bc"},"challengeLock":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x057e28dce74a7e6d03e4a3cdc90ec4d096c38ebea880dd6b0d6f549003bb6bc7"},"typeHash":"0xb580c581b8789dbd3add6c8d2efc63af3590b1a272d950e7dd1c2f64f5b425b5"},"l1Sudt":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x4db75e03349f4f2ec792476035dd1b7376c683130f7e2e74024be2d9ee064511"},"typeHash":"0xc5e5dcf215925f7ef4dfaf5f4b4f105bc321c02776d6e7d52a1db3fcd9d011a4"},"l2Sudt":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0xc1fc7c7a3c09ed5e637128fadf60a4d87a534011a0773e83f03ac6ff56c329ff"},"typeHash":"0x6900864b01b49abad8c683cbf8641e3df2c12e362596eb8bab88ce99ff3b6f92"},"omniLock":{"script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x1b8572b16c07f46a0efed623aea6de05d45985b9a7c1b0b52276da5d9f9615b7"},"typeHash":"0x79f90bb5e892d80dd213439eeab551120eb417678824f282b4ffb5f21bad2e1e"}},"eoaScripts":{"eth":{"typeHash":"0x8d6af470fd57d1afe543751eed58afef310ec659d565319f448bcc765176dcdc","script":{"code_hash":"0x00000000000000000000000000000000000000000000000000545950455f4944","hash_type":"type","args":"0x414e3a6b031cd3c18333fd071d9dd15666233b3889f45c5e9186492daf7fd8f0"}}},"backends":{"sudt":{"validatorCodeHash":"0xe9f21f08e1a09748b701be259c17e1dd9f1accaee23e2c6e0609916d308a1b4d","generatorCodeHash":"0x34f8d804e2173c076495e78458ab9031c769c36062fede0cb35e7762db04224d","validatorScriptTypeHash":"0x6900864b01b49abad8c683cbf8641e3df2c12e362596eb8bab88ce99ff3b6f92"},"meta":{"validatorCodeHash":"0x65c15510932d7ca6622e7a9c84febc4861d50423b4c1dbeb1bc33ad79312dd57","generatorCodeHash":"0xbf9a4e7cdb07f0234efc5ad0631d3797badf0cc7ce6d7d938388a855e9f7eb3f","validatorScriptTypeHash":"0x953a0992e397c76eb0e1cd16ff8998a59c00f9280586fda31e3fe1d99bd6707a"},"polyjuice":{"validatorCodeHash":"0x320f87eb51c6f4bbcbcbed5e31e97e8c4580d49965259519476ec2985b7fa0b6","generatorCodeHash":"0x15b4b918955bb92f036abfc350b9587d1983fba8f94de85973a4c0e70f98b1c1","validatorScriptTypeHash":"0x7907a044582d425a584bd2759ff1855dfc66f0c5391e1be932c9716f58742b15"},"ethAddrReg":{"validatorCodeHash":"0x57553b7cb7736307865ace9250c492460f12291e685d81e8d0a2e033e5582a00","generatorCodeHash":"0x96ac862b9791c0d1cb1ccf85c1f1d60b1e359a1288d52fad6987b9fc53b0b289","validatorScriptTypeHash":"0xdd2c6955089f34785ea049d62335cb06982e665d73687626e6b4af09b8a7774b"}},"accounts":{"polyjuiceCreator":{"id":"0x4","scriptHash":"0x6e55e1014b011dfa2aa12f4833dcf347d966c5eb5106196f52f945fd048b73a2"},"ethAddrReg":{"id":"0x2","scriptHash":"0xc88f2395683ed4a69141d76f8fc6c612d10f7d8c964f7ddcd0b2182fda545d73"},"defaultFrom":{"id":"0x3","scriptHash":"0x130c25347beb6744ea1c9eb75a6a2b8dd23c04f83402cfdeb81f41915d968544"}},"chainId":"0x315de"}}}
```

 eth_getTransactionReceipt ' transactionIndex !=  log.transactionIndex
```
 curl --location --request POST 'https://godwoken-alphanet-v1.ckbapp.dev' \
--header 'Content-Type: application/json' \
--data-raw '{
        "jsonrpc":"2.0",
        "method":"eth_getTransactionReceipt",
        "params":[
                "0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113"
        ],
        "id":1
}'
{"jsonrpc":"2.0","id":1,"result":{"transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x1","gasUsed":"0x135a6","cumulativeGasUsed":"0x26b4c","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000","logs":[{"address":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x0","transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","data":"0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","logIndex":"0x0","topics":[],"removed":false},{"address":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x0","transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","data":"0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","logIndex":"0x1","topics":["0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20"],"removed":false},{"address":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x0","transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","data":"0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","logIndex":"0x2","topics":["0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","0x00000000000000000000000070997970c51812dc3a010c7d01b50e0d17dc79c8"],"removed":false},{"address":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x0","transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","data":"0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","logIndex":"0x3","topics":["0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","0x00000000000000000000000070997970c51812dc3a010c7d01b50e0d17dc79c8","0x0000000000000000000000000000000000000000000000000000000000420042"],"removed":false},{"address":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","blockHash":"0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85","blockNumber":"0x2d355","transactionIndex":"0x0","transactionHash":"0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113","data":"0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","logIndex":"0x4","topics":["0x50cb9fe53daa9737b786ab3646f04d0150dc50ef4e75f59509d83667ad5adb20","0x00000000000000000000000070997970c51812dc3a010c7d01b50e0d17dc79c8","0x0000000000000000000000000000000000000000000000000000000000420042","0x0000000000000000000000000000000000000000000000000000000000420042"],"removed":false}],"contractAddress":"0xb2dd30e01692520bbc59cf9095576dcf2b651e2c","status":"0x1","from":"0x70997970c51812dc3a010c7d01b50e0d17dc79c8","to":null}}
```

### expected 
eth_getTransactionReceipt 's  transactionIndex == log.transactionIndex

**Comment by keroro520** [comment]
```
eth_getTransactionByBlockHashAndIndex(
  "0xbaef067a8b67ca9e033446d3c33ae14bfef088489487a8d96b373a60483cca85",
  "0x1"
)

==> {"hash"： "0xc536c9f73cb0be80afb338191234d3dc5a756d599bc71daecefcf668deb95113"}
```

The log.transactionIndex is wrong.

**Comment by Flouse** [comment]
fixed in [Web3 v1.6.3+](https://github.com/nervosnetwork/godwoken-web3/releases/tag/v1.6.3)
